### PR TITLE
Add GolangCI via Makefile + Presubmit.sh + golangci lint fixes

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Address potential bugs picked up by the golangci linter. Enable golang lint checks in our CI tests.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/ok-to-test

File: .golangci.yml
Comment (AndrienkoAleksandr) [reply .golangci.yml]: Nice catch, @sayan-biswas. Thanks.

**Review by khrm — COMMENTED** [review]
/test pull-tekton-results-integration-tests

**Review by khrm — COMMENTED** [review]
/approve

**Comment by khrm** [comment]
@enarha @sayan-biswas I think we can merge this now. Please give lgtm here.

File: cmd/api/main.go
Comment (adambkaplan) [inline cmd/api/main.go:68]: Please add a comment justifying why we are skipping the error check.

File: cmd/watcher/main.go
Comment (adambkaplan) [inline cmd/watcher/main.go:53]: Please add a comment explaining this is not a hard-coded secret:

```suggestion
        // This is a fixed path which does not contain a hard-coded secret or credential
	podTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token" //nolint:gosec
```

**Review by adambkaplan — COMMENTED** [review]
@AndrienkoAleksandr great work here! A lot of items uncovered by the linter.

My only ask before we merge is that commits are squashed, and that the commit message briefly describes the kinds of changes made to get the linter happy. You do not need to go into deep details of every code change.

**Comment by khrm** [comment]
Yes, the last three commits can be merged in the same. 

The first commit from me to enable the GolangCI.
And. one for all the fixes that we did to make it work.

**Review by sayan-biswas — APPROVED** [review]

**Comment by AndrienkoAleksandr** [comment]
@adambkaplan, @khrm, @sayan-biswas I handled code review feedback, squashed my commits  and rebased pr.

**Review by khrm — COMMENTED** [review]
/lgtm

**Comment by enarha** [comment]
The change is LGTM. I'll approve once the test is fixed.

**Comment by enarha** [comment]
Actually one comment. So now we are running a bunch of linters in CI. Shouldn't we document the steps to run those locally before submitting a PR to eliminate issues called only in CI which is more expensive in terms of resources?

**Comment by enarha** [comment]
The failing test also fails in main when running locally. And it's failing pretty consistently.

**Comment by khrm** [comment]
Issue isn't with results, I think it's with a pr merged in pipelines. Let me fixed that.

**Pull request merged**
